docs/ionic/authentication.md

```diff
@@ -0,0 +1,94 @@
+# STEP 1: install this plugin: 
+```
+1) ionic cordova plugin add cordova-universal-links-plugin
+2) ionic cordova plugin add cordova-plugin-buildinfo
+3) ionic cordova plugin add cordova-plugin-browsertab
+4) ionic cordova plugin add cordova-plugin-inappbrowser
+5) ionic cordova plugin add cordova-plugin-customurlscheme (for ios)
+```
+
+# STEP 2: Add Firebase to your Ionic
+
+  STEP 2.1:  [setup firebase to angular project] 
+  (https://github.com/angular/angularfire2/blob/master/docs/install-and-setup.md)
+
+  STEP 2.2: To set up an Android app, go to [Firebase Console](https://console.firebase.google.com/) then 
+  Click **Add Firebase to your Android app** and follow the setup steps.
+
+# STEP 3: Set up Firebase Authentication for Cordova ( sumary from [firebase instruction](https://firebase.google.com/docs/auth/web/cordova))
+
+ STEP 3.1: In the Firebase console, open the **Dynamic Links** section at bottom left panel, setup by they instruction
+
+ STEP 3.2: add this to config.xml at root level of project:
+```
+     <universal-links>
+        <!-- this is dynamic link created in firebase -->
+        <host name="zm4e4.app.goo.gl" scheme="https" />
+        <!-- this is your firebase app link -->
+        <host name="routing-aadd4.firebaseapp.com" scheme="https">
+            <path url="/__/auth/callback" />
+        </host>
+    </universal-links>
+    <!-- for android -->
+    <preference name="AndroidLaunchMode" value="singleTask" />
+```
+
+  STEP 3.3: make sure your `<widget id="com.yourandroid.id" ... >` the same with android app's id you 
+  added in firebase at STEP 2.2.
+
+# STEP 4: add login code:
+at login.service.ts add this function: 
+```
+import { AngularFireAuth } from 'angularfire2/auth';
+import * as firebase from 'firebase/app';
+import AuthProvider = firebase.auth.AuthProvider;
+
+export class AuthService {
+        private user: firebase.User;
+	constructor(public afAuth: AngularFireAuth) {
+		afAuth.authState.subscribe(user => {
+			this.user = user;
+		});
+	}
+  
+  signInWithFacebook() {
+		console.log('Sign in with google');
+		return this.oauthSignIn(new firebase.auth.FacebookAuthProvider());
+	}
+
+  signInWithGoogle() {
+		console.log('Sign in with google');
+		return this.oauthSignIn(new firebase.auth.GoogleAuthProvider());
+	}
+
+        private oauthSignIn(provider: AuthProvider) {
```

Comment:
The code formatting is off in this sample. Try pasting it into an editor to clean it up for you and then paste it back in here.